frontend/src/slices/chat/chat.slice.ts

```diff
@@ -0,0 +1,69 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import {
+  FIRST_INDEX,
+  LAST_INDEX,
+  MOCK_MESSAGE,
+} from '#libs/constants/constants.js';
+import {
+  type ChatMessage,
+  type ChatState,
+  type Message,
+} from '#libs/types/types.js';
+
+import { newMessage } from './actions.js';
+
+const initialState: ChatState = { messages: [MOCK_MESSAGE] };
+
+type StateInfo = ChatMessage & {
+  state: ChatState;
+};
+
+const isSameUserLastMessage = (state: ChatState, sender: string): boolean => {
+  return state.messages.at(LAST_INDEX)?.sender === sender;
+};
+
+const appendLastMessage = (state: ChatState, message: Message): void => {
+  const updatedLastMessage = {
+    ...state.messages.at(LAST_INDEX),
+  } as ChatMessage;
+  updatedLastMessage.messages.push(message);
+  state.messages = [
+    ...state.messages.slice(FIRST_INDEX, LAST_INDEX),
+    updatedLastMessage,
+  ];
+};
+
+const addMessageToState = ({
+  state,
+  id,
+  sender,
+  messages,
+}: StateInfo): void => {
```

Comment:
i think all this logic can be removed

Reply:
actually, that is what I wanted to ask) @Xxandder and I are working on the chat slice that will take the backend response into consideration

Reply:
ok, let's remove it